doc/holochain_101/src/writing_development_kit.md

```diff
@@ -1 +1,100 @@
+  a. use a 16 bit integer for the pointers `offset`  
+  b. use a 16 bit integer for the pointers `length`
+9. join the pointers into a single 32 bit integer  
+  a. high bits are `offset`  
+  b. low bits are `length`
+10. call the native function with that 32 bit integer and assign the result to another 32 bit integer  
+  a. e.g. `encoded_alloc_of_result = hc_commit_entry(encoded_alloc_of_input)`
+11. deconstruct that 32 bit integer into two variables  
+  a. use a 16 bit integer for the pointers `offset`  
+  b. use a 16 bit integer for the pointers `length`
+12. read string data from memory at the `offset` address
+13. deallocate the memory
+14. deserialize the string to JSON if JSON is expected
+
+That looks like a lot of steps, but most of this code can be shared for the various functions throughout the Development Kit, leaving implementations to be as little as 5 lines long. Basically, the process inverts at the point of the native function call.
+
+#### WASM Single Page Stack
+
+TODO
+
+### App Globals
+
+When writing Zome code, it is common to need to reference aspects of the context it runs in, such as the active user/agent, or the DNA hash of the app. Holochain exposes certain values through to the Zome, though it does so natively by way of the `hc_init_globals` function mentioned. Again, to simplify the developer experience, it is valuable to expose these values as constants through to the developer.
+
+This is done by calling `hc_init_globals` with an input value of 0. The result of calling the function is a 32 bit integer which represents the memory location of a serialized JSON object containing all the app global values. Fetch the result from memory, and deserialize the result back into an object. If appropriate, set those values as exports for the Development Kit. For example, in Rust, values become accessible in Zomes using `hdk::APP_NAME`. It's recommended to use all capital letters for the export of the constants, but as they are returned as keys on an object from `hc_init_globals` they are in lower case. The object has the following values:
+- app_name
+- app_dna_hash
+- app_agent_id_str
+- app_agent_key_hash
+- app_agent_initial_hash
+- app_agent_latest_hash
+
+See the [API global variables](/zome/api_globals.html) page for details on what these are.
+
+### Publish It and Get In Touch
+
+If you've made it through the process so far, good work. The community is an important part of the success of any project, and Holochain is no different. If you're really proud of your work, get in touch with the team in their [chat server](https://chat.holochain.net/appsup/channels/hc-core), and let them know you're working on it. Even ask for help. This book could be updated to include links to other HDKs. Whether you would like to, or you'd like the team to, the HDK should be published to the primary package manager in use for the language. For example, RubyGems for Ruby or npm for nodejs.
```

Comment:
I'm mindful of what language we use... I was thinking maybe to try not to tie the book too closely to the organization... make it more about the open source-ness and the community. Which is why I used "their" and "they" not "ours" and "us".

Reply:
i agree. i try to do the same. it's often possible to avoid those words entirely.

Reply:
I appreciate the sentiment here, for sure. It's also a little contradictory to be producing an early guidebook and then use a voice not associated with the org. I think with docs that show our explicit attempt to solicit community support, it's not an awful thing to write in awareness of where the project is at. People may actually feel more comfortable in the early days knowing that the guide they're reading was written by folks who really know what they're talking about.

Reply:
Good point also. I think I will just revise to avoid the words entirely, as you say David. I think it keeps it nicely neutral